src/Collection/CollectionInterface.php

```diff
@@ -578,6 +653,27 @@ public function sample(int $length = 10): CollectionInterface;
      */
     public function take(int $length = 1, int $offset = 0): CollectionInterface;
 
+    /**
+     * Returns a new collection containing a subset of elements for a specified page.
+     *
+     * This method paginates the collection by dividing it into pages, where each page
+     * contains a maximum of $perPage elements. The $page parameter indicates the
+     * desired page number, and $perPage specifies the number of elements per page.
+     *
+     * ### Example:
+     *
+     * ```
+     * $collection = new Collection([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+     * $pagedCollection = $collection->paginate(page: 2, perPage: 3);
+     * // $pagedCollection will contain elements [4, 5, 6] for page 2.
+     * ```
+     *
+     * @param int<1, max> $page The page number to retrieve (1-based index).
+     * @param int<1, max> $perPage The number of elements per page.
+     * @return self A new collection containing elements for the specified page.
+     */
+    public function paginate(int $page, int $perPage): CollectionInterface;
```

Comment:
If the dataset is already in memory, why do you need to paginate it? Collection has another method -- `chunk` which works similar to this one as well.

Reply:
As mentioned, the collection interface already includes the `take(length, offset)` method, but it may not be immediately intuitive for pagination purposes. Furthermore, `paginate` method leverages the `take` method and doesn't add more complexity to maintain the code.

Reply:
> Furthermore, paginate method leverages the take method and doesn't add more complexity to maintain the code.

More methods = more complexity though. Having multiple methods that do similar things adds complexity for users as well. They have more methods to understand and make decisions between. So while `take()` isn't ideal for pagination, I not sure we should be encouraging pagination of data that is already in memory.

Reply:
> I not sure we should be encouraging pagination of data that is already in memory.

I share the same opinion.